src/impl_methods.rs

```diff
@@ -1826,12 +1826,18 @@ impl<A, S, D> ArrayBase<S, D> where S: Data<Elem=A>, D: Dimension
     /// function and initial value `init`.
     ///
     /// Return the result as an `Array`.
+    ///
+    /// **Panics** if `axis` is out of bounds.
     pub fn fold_axis<B, F>(&self, axis: Axis, init: B, mut fold: F)
         -> Array<B, D::Smaller>
         where D: RemoveAxis,
               F: FnMut(&B, &A) -> B,
               B: Clone,
     {
+        if axis.index() >= self.ndim() {
+            panic!("ndarray: axis {} out of bounds in array dim {} in fold_axis",
```

Comment:
```suggestion
            panic!("ndarray: axis {} out of bounds in array with ndim={} in fold_axis",
```

or

"ndarray: axis {} out of bounds in array of dim {} in fold_axis"

Reply:
Thanks, using `array of dim`